src/lib/plugins/log.js

```diff
@@ -1,20 +1,39 @@
 const fs = require('fs')
-const timeStarted = Math.floor(new Date() / 1000).toString()
+const timeStarted = Math.floor(Date.now() / 1000).toString()
 const path = require('path')
-const mkdirp = require('mkdirp')
 const moment = require('moment')
 const colors = require('colors')
 
-const readline = require('readline')
-const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
-})
+const isInNode = typeof process !== 'undefined' && !process.browser && process.platform !== 'browser'
 
-rl.setPrompt('> ')
-rl.prompt(true)
+const _servers = []
+
+/** @type {typeof import("readline") | undefined} */
+let readline
+/** @type {import("readline").Interface | undefined} */
+let rl
+if (isInNode) {
+  import(/* webpackIgnore: true */ 'exit-hook').then((hook) => {
+    hook.default(() => {
+      for (const serv of _servers) {
+        serv.log('Server shutting down...')
+        serv.quit()
+      }
+    })
+  })
+  readline = require('readline')
+  rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  })
+
+  rl.setPrompt('> ')
+  rl.prompt(true)
+}
 
 module.exports.server = function (serv, settings) {
+  _servers.push(serv)
```

Comment:
What is _servers ?

Reply:
I thought it was possible to create multiple servers. However, now I understand that I could just inline exit-hook handler here.